sztp-agent/pkg/secureagent/daemon.go

```diff
@@ -56,60 +57,69 @@ func (a *Agent) RunCommandDaemon() error {
 
 func (a *Agent) performBootstrapSequence() error {
 	var err error
-	if a.GetBootstrapURL() == "" {
+	// check if empty
+	if len(a.GetBootstrapURL()) == 1 && a.GetBootstrapURL()[0] == "" {
```

Comment:
something doesn't feel right here...
1. if user gave me URL to run on, run on it, don't check dhcp leases or network manager
2. if user gave me leases file, parse it, don;t check network manager
3. check network manager